File: OrganizationService/src/main/java/com/slavacom/organizationservice/entity/ProjectHistory.java
package com.slavacom.organizationservice.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.Instant;
import java.util.UUID;

@Getter
@Setter
@Entity
@Table(name = "project_history")
public class ProjectHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id", nullable = false)
    private UUID id;

    @Column(name = "project_id", nullable = false)
    private UUID projectId;

    @Column(name = "changed_by")
    private UUID changedBy;

    @Column(name = "action", nullable = false)
    private String action;

    @Column(name = "changes", columnDefinition = "TEXT")
    private String changes;

    @CreationTimestamp
    @Column(name = "changed_at", nullable = false, updatable = false)
    private Instant changedAt;
}

File: OrganizationService/src/main/java/com/slavacom/organizationservice/entity/ProjectHistory.kt
package com.slavacom.organizationservice.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.CreationTimestamp
import java.time.Instant
import java.util.UUID

@Entity
@Table(name = "project_history")
open class ProjectHistory {
    @field:Id
    @field:GeneratedValue(strategy = GenerationType.UUID)
    @field:Column(name = "id", nullable = false)
    open var id: UUID? = null

    @field:Column(name = "project_id", nullable = false)
    open var projectId: UUID? = null

    @field:Column(name = "changed_by")
    open var changedBy: UUID? = null

    @field:Column(name = "action", nullable = false)
    open var action: String? = null

    @field:Column(name = "changes", columnDefinition = "TEXT")
    open var changes: String? = null

    @field:CreationTimestamp
    @field:Column(name = "changed_at", nullable = false, updatable = false)
    open var changedAt: Instant? = null
}

